src/permutat.cc

```diff
@@ -1125,6 +1125,40 @@ UInt LargestMovedPointPerm(Obj perm)
         return LargestMovedPointPerm_<UInt4>(perm);
 }
 
+/****************************************************************************
+**
+*F  SmallestMovedPointPerm( <perm> ) smallest point moved by perm
+**
+**  'SmallestMovedPointPerm' implements 'SmallestMovedPoint' for internal
+**  permutations.
+*/
+
+// Import 'Infinity', as a return value for the identity permutation
+static Obj Infinity;
+
+template <typename T>
+static inline Obj SmallestMovedPointPerm_(Obj perm)
+{
+    const T * ptPerm = CONST_ADDR_PERM<T>(perm);
+    UInt      deg = DEG_PERM<T>(perm);
+
+    for (UInt i = 1; i <= deg; i++) {
+        if (ptPerm[i - 1] != i - 1)
```

Comment:
Why don't you instead do the loop from `i = 0; i < deg` so that you don't have to do `i - 1` twice on this line for every iteration of the loop? Then on the following line you'd just do
```
return INTOBJ_INT(i + 1);
```

Reply:
The boring answer to your question is "LargestMovedPointPerm" does that, and I kept the two as close as possible, as any differences might raise questions of "why are they different".

Of course, the other option is to clean them both up (which I think I'll do, in another commit).

Reply:
Actually, looking through, basically everywhere in permutat.cc uses this 'ptr[i-1] = i-1' style, so it would seem sensible to clean them all up, instead of just one, and I'd do that in a future PR.

Reply:
Happy for that to be done separately. I made #3739 to remind you/us to do it.